packages/components/src/focal-point-picker/types.ts

```diff
@@ -0,0 +1,84 @@
+/**
+ * External dependencies
+ */
+import type {
+	CSSProperties,
+	ReactEventHandler,
+	Ref,
+	VideoHTMLAttributes,
+} from 'react';
+
+/**
+ * Internal dependencies
+ */
+import type { BaseControlProps } from '../base-control/types';
+
+export type FocalPoint = Record< FocalPointAxis, number >;
+export type FocalPointAxis = 'x' | 'y';
+
+export type FocalPointPickerProps = Pick<
+	BaseControlProps,
+	'help' | 'hideLabelFromVision' | 'label'
+> & {
+	/**
+	 * Autoplays HTML5 video. This only applies to video sources (`url`).
+	 *
+	 * @default true
+	 */
+	autoPlay?: boolean;
+	/**
+	 * Callback which is called when the focal point changes.
+	 */
+	onChange: ( value: FocalPoint ) => void;
+	onDrag?: ( value: FocalPoint, event: MouseEvent ) => void;
+	/**
+	 * Callback which is called at the end of drag operations.
+	 */
+	onDragEnd?: () => void;
+	/**
+	 * Callback which is called at the start of drag operations.
+	 */
+	onDragStart?: (
+		value: FocalPoint,
+		event: React.MouseEvent< Element, MouseEvent >
+	) => void;
+	/**
+	 * Function which is called before internal updates to the value state.
+	 * It receives the upcoming value and may return a modified one.
+	 */
+	resolvePoint?: ( point: FocalPoint ) => FocalPoint | undefined;
```

Comment:
Yeah that makes sense. I think I just hastily copied from the inferred type, which included `undefined` as a return value because `resolvePoint` itself was optional. Fixed 👍